### PR TITLE
(SERVER-893) Apply pl-clojure-style recommendations

### PR DESCRIPTION
**Pull request by justinstoller**
This is for review. This uses the commits in https://github.com/puppetlabs/trapperkeeper/pull/226 but with style updates for service and fnk here: https://github.com/puppetlabs/pl-clojure-style/pull/3.

This is so that folks reviewing the above PRs can see the output of the changes and compare them to the diffs generated in https://github.com/puppetlabs/trapperkeeper/pull/227 .

_If_ everything's copacetic, once the pl-clojure-style PR is merged I'll update this bump the submodule to use it.

### CODE REVIEW

**Comment by justinstoller** [comment]
Updated with https://github.com/puppetlabs/pl-clojure-style/pull/3/commits/61e75c4bebad0c9b13b0c08143eac770ab643d19

**Comment by KevinCorcoran** [comment]
This looks ready for one more re-work.

**Comment by justinstoller** [comment]
Updated.

File: examples/java_service/src/clj/java_service_example/java_service.clj [hunk 1]
```diff
@@ -8,20 +8,20 @@
   (meaning-of-life-fn [this]))
 
 (defservice java-service
-            JavaService
-            []
+  JavaService
+  []
             ;; Service functions are implemented in a java `ServiceImpl` class
```
Comment (KevinCorcoran) [inline examples/java_service/src/clj/java_service_example/java_service.clj]: There's now too much indentation on this line.  I wonder if there's a way to get cljfmt to fix these automatically.

**Comment by KevinCorcoran** [comment]
🙈 except for the comment about indentation on comments, although I'm guessing we won't be able to get cljfmt to fix those for us.

**Comment by justinstoller** [comment]
@KevinCorcoran from a quick look into it, the underlying library cljfmt uses (rewrite-clj) supports managing comments, and cljfmt has some comment aware code, but doesn't expose managing comments to end users.

File: examples/java_service/src/clj/java_service_example/java_service.clj (hunk 1 above)
Comment (justinstoller) [inline examples/java_service/src/clj/java_service_example/java_service.clj]: manually tweaked in 4e59be5

**Comment by rlinehan** [comment]
:+1: skimmed, everything looks reasonable, ran `lein cljfmt check` on it and it turned up nothing, :boom:

**Pull request merged**